numpy/polynomial/polynomial.py

```diff
@@ -654,6 +657,10 @@ def polyint(c, m=1, k=[], lbnd=0, scl=1, axis=0):
         raise ValueError("The order of integration must be non-negative")
     if len(k) > cnt:
         raise ValueError("Too many integration constants")
+    if np.ndim(lbnd) != 0:
+        raise TypeError("lbnd must be a scalar.")
+    if np.ndim(scl) != 0:
+        raise TypeError("scl must be a scalar.")
     if iaxis != axis:
         raise ValueError("The axis must be integer")
```

Comment:
This should be `operator.index`, not `int(axis) == axis`

Reply:
In fact, this line should just be removed, and the `normalize_axis_index` below allowed to throw a more useful error.

Reply:
But that's not part of this PR, so don't worry about it